web/controllers/friends_controller.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Entice.Web.FriendsController do
+  use Entice.Web.Web, :controller
+  alias Entice.Web.Queries
+
+  plug :ensure_login
+
+  @doc "Returns all friends of connected account."
+  def index(conn, _params) do
+    id = get_session(conn, :client_id)    
+    {:ok, friends} = Entice.Web.Client.get_friends(id)
+    
+    friends = for friend <- friends, do: friend.friend_account.id
+    conn |> json ok%{
+      message: "All friends",
+      friends: friends}
+  end
+
+
+  @doc "Adds friend :id to friends list of connected account."
+  def create(conn, _params) do
+    session_id = get_session(conn, :client_id)
+    {:ok, acc} = Client.get_account(session_id)
+    account_id = acc.id
+    friend_account_id = conn.params["friend_account_id"]
+
+    friend_account = Entice.Web.Repo.get(Entice.Web.Account, friend_account_id)
+    
+    #Must refactor this
+    result = case friend_account do
+      nil -> error(%{message: "This account does not exist."})
+      _ ->
+        result = case friend_account.id do
```

Comment:
Also, you dont need to re-assign result all the time - the value of the last evaluated expression will get carried to the topmost level ;)